.github/workflows/integration_test_charm.yaml

```diff
@@ -412,21 +412,41 @@ jobs:
       - name: juju status
         timeout-minutes: 1
         if: ${{ success() || (failure() && steps.tests.outcome == 'failure') }}
-        run: juju status --color --relations | tee ~/logs/juju-status.txt
+        run: |
+          MODELS=$(juju models --format json | jq -r '.models[] | select(."short-name" != "controller") | ."short-name"' | sed ':a;N;$!ba;s/\n/ /g')
+          for MODEL in $MODELS
+          do
+            juju status --model="$MODEL" --color --relations | tee "$HOME/logs/juju-status-$MODEL.txt"
+            printf "\n"
+          done
       - name: juju debug-log
         timeout-minutes: 3
         if: ${{ success() || (failure() && steps.tests.outcome == 'failure') }}
         # TODO future improvement: show models/ logs in juju-debug-log.txt to match `juju debug-log`
         run: |
+          MODELS=$(juju models --format json | jq -r '.models[] | select(."short-name" != "controller") | ."short-name"' | sed ':a;N;$!ba;s/\n/ /g')
           if [[ '${{ inputs.cloud }}' == 'lxd' ]]
           then
             mkdir ~/logs/models/
             juju scp --model controller 0:/var/log/juju/models/* ~/logs/models/
             juju scp --model controller 0:/var/log/juju/logsink.log ~/logs/controller-logsink.log
-            convert-logsink-to-debug-log ~/logs/controller-logsink.log ~/logs/juju-debug-log.txt
-            cat ~/logs/juju-debug-log.txt
+            ORIGINAL_MODEL=$(juju switch)
+            for MODEL in $MODELS
+            do
+              juju switch --quiet "$MODEL"
+              convert-logsink-to-debug-log ~/logs/controller-logsink.log "$HOME/logs/juju-debug-log-$MODEL.txt"
+              printf "Model: %s\n" "$MODEL"
+              cat "$HOME/logs/juju-debug-log-$MODEL.txt"
+              printf "\n"
+            done
+            juju switch --quiet "$ORIGINAL_MODEL"
```

Comment:
```suggestion
            ORIGINAL_MODEL=$(juju switch)
            for MODEL in ${MODELS[@]}; do
              juju switch --quiet "${MODEL}"
              convert-logsink-to-debug-log ~/logs/controller-logsink.log ~/logs/"juju-debug-log-${MODEL}.txt"
              echo -e "\nModel: ${MODEL}"
              cat ~/logs/"juju-debug-log-${MODEL}.txt"
            done
            juju switch --quiet "${ORIGINAL_MODEL}"
```

Reply:
Thanks for the improvement, Alex! Fixed in the same commit commented on https://github.com/canonical/data-platform-workflows/pull/195#discussion_r1646513966.